hail_search/queries/base.py

```diff
@@ -644,25 +644,28 @@ def _parse_intervals(self, intervals, gene_ids=None, **kwargs):
 
         raw_intervals = intervals
         if self._should_add_chr_prefix():
-            intervals = [
-                f'[chr{interval.replace("[", "")}' if interval.startswith('[') else f'chr{interval}'
-                for interval in (intervals or [])
-            ]
-
-        if is_x_linked:
-            reference_genome = hl.get_reference(self.GENOME_VERSION)
-            intervals = (intervals or []) + [reference_genome.x_contigs[0]]
+            intervals = [[f'chr{interval[0]}', *interval[1:]] for interval in (intervals or [])]
 
         if len(intervals) > MAX_GENE_INTERVALS and len(intervals) == len(gene_ids or []):
-            return []
+            super_intervals = defaultdict(lambda: (1e9, 0))
+            for chrom, start, end in intervals:
+                super_intervals[chrom] = (min(super_intervals[chrom][0], start), max(super_intervals[chrom][1], end))
+            intervals = [(chrom, start, end) for chrom, (start, end) in super_intervals.items()]
```

Comment:
this is the actual performance change in this PR

Reply:
this is clever I like it!